server/command.go

```diff
@@ -188,6 +336,116 @@ func (p *Plugin) executeCommandDeleteWelcome(args *model.CommandArgs) {
 	p.postCommandResponse(args, "welcome message has been deleted")
 }
 
+func (p *Plugin) executeCommandSetTeamWelcome(args *model.CommandArgs) {
+	doesUserHavePrivileges := p.isSystemOrTeamAdmin(args, args.UserId, args.TeamId)
+	if !doesUserHavePrivileges {
+		return
+	}
+
+	// Fields will consume ALL whitespace, so plain re-joining of the
+	// parameters slice will not produce the same message
+	message := strings.SplitN(args.Command, "set_team_welcome", 2)[1]
+	message = strings.TrimSpace(message)
+
+	key := makeTeamWelcomeMessageKey(args.TeamId)
+	if appErr := p.API.KVSet(key, []byte(message)); appErr != nil {
+		p.postCommandResponse(args, "error occurred while storing the welcome message for the team: `%s`", appErr)
+		return
+	}
+
+	p.postCommandResponse(args, "stored the team welcome message:\n%s", message)
+}
+
+func (p *Plugin) executeCommandDeleteTeamWelcome(args *model.CommandArgs) {
+	doesUserHavePrivileges := p.isSystemOrTeamAdmin(args, args.UserId, args.TeamId)
+	if !doesUserHavePrivileges {
+		return
+	}
+
+	key := makeTeamWelcomeMessageKey(args.TeamId)
+	_, appErr := p.GetTeamWelcomeMessageFromKV(args.TeamId)
+
+	if appErr != nil {
+		p.postCommandResponse(args, "error occurred while retrieving the welcome message for the team: `%s`", appErr)
+		return
+	}
+
+	if appErr := p.API.KVDelete(key); appErr != nil {
+		p.postCommandResponse(args, "error occurred while deleting the welcome message for the team: `%s`", appErr)
+		return
+	}
+
+	p.postCommandResponse(args, "team welcome message has been deleted")
+}
+
+func (p *Plugin) executeCommandGetTeamWelcome(args *model.CommandArgs) {
+	doesUserHavePrivileges := p.isSystemOrTeamAdmin(args, args.UserId, args.TeamId)
+	if !doesUserHavePrivileges {
+		return
+	}
+
+	data, appErr := p.GetTeamWelcomeMessageFromKV(args.TeamId)
+	if appErr != nil {
+		p.postCommandResponse(args, "error occurred while retrieving the welcome message for the team: `%s`", appErr)
+		return
+	}
+
+	// retrieve team name through the teamid
+	team, err := p.API.GetTeam(args.TeamId)
+	if err != nil {
+		p.postCommandResponse(args, err.Error())
+		return
+	}
+
+	if data == nil {
+		for _, message := range p.getWelcomeMessages() {
+			if message.TeamName == team.Name {
+				if err := p.previewWelcomeMessage(team.Name, args, *message); err != nil {
+					p.postCommandResponse(args, "error occurred while processing greeting for team `%s`: `%s`", team.Name, err)
+				}
+				return
+			}
+		}
+		// if KV do not have message, and Config.json does not have message, then there is no message. Display Error case.
+		p.postCommandResponse(args, unsetMessageError)
+		return
+	}
+	p.postCommandResponse(args, string(data))
+}
+
+func (p *Plugin) checkCommandPermission(args *model.CommandArgs) bool {
```

Comment:
I'm inclined to change the signature to 
```suggestion
func (p *Plugin) checkCommandPermission(args *model.CommandArgs) (bool, err) {
```

as it's more idiomatic to return an error instead of handling it in the function directly.

Reply:
New PR Will contain the signature fix, as well as the commented out section for the town square check! 

I also fixed the signature of the `validatePreviewPriviledges`.

Those changes should be seen with the next push to this PR! :)